pajbot/modules/wide_emote_limit.py

```diff
@@ -0,0 +1,121 @@
+import logging
+
+from pajbot.managers.handler import HandlerManager
+from pajbot.modules import BaseModule, ModuleSetting
+
+log = logging.getLogger(__name__)
+
+
+class WideEmoteLimitModule(BaseModule):
+    ID = __name__.split(".")[-1]
+    NAME = "Wide Emote Limit"
+    DESCRIPTION = "Times out users who post too many wide 7TV or FFZ emotes"
+    CATEGORY = "Moderation"
+    SETTINGS = [
+        ModuleSetting(
+            key="max_emotes",
+            label="Maximum number of wide emotes that can be posted",
+            type="number",
+            required=True,
+            placeholder="",
+            default=15,
+            constraints={"min_value": 1, "max_value": 167},
+        ),
+        ModuleSetting(
+            key="max_width",
+            label="Maximum width of emotes in pixels. Emotes exceeding this width will be counted as wide",
+            type="number",
+            required=True,
+            placeholder="",
+            default=128,
+            constraints={"min_value": 112, "max_value": 1000},
+        ),
+        ModuleSetting(
+            key="bypass_level",
+            label="Level to bypass module",
+            type="number",
+            required=True,
+            placeholder="",
+            default=420,
+            constraints={"min_value": 100, "max_value": 1000},
+        ),
+        ModuleSetting(
+            key="moderation_action",
+            label="Moderation action to apply",
+            type="options",
+            required=True,
+            default="Timeout",
+            options=["Delete", "Timeout"],
+        ),
+        ModuleSetting(
+            key="timeout_duration",
+            label="Timeout duration (if moderation action is timeout)",
+            type="number",
+            required=True,
+            placeholder="",
+            default=60,
+            constraints={"min_value": 1, "max_value": 1209600},
+        ),
+        ModuleSetting(
+            key="allow_subs_to_bypass",
+            label="Allow subscribers to bypass",
+            type="boolean",
+            required=True,
+            default=False,
+        ),
+        ModuleSetting(
+            key="enable_in_online_chat", label="Enabled in online chat", type="boolean", required=True, default=True
+        ),
+        ModuleSetting(
+            key="enable_in_offline_chat", label="Enabled in offline chat", type="boolean", required=True, default=True
+        ),
```

Comment:
Can we replace `timeout_duration` with an online and offline timeout duration? Some streamers may want a different timeout length in offline chat compared to online chat.